packages/coil-extension/src/background/services/AuthService.ts

```diff
@@ -31,6 +70,21 @@ export class AuthService extends EventEmitter {
 
   private _op: Promise<string | null> | null = null
 
+  /*
+  TODO: manifest version 3 and background workers ?
+  If the token is issued more than one day ago, refresh it, such that the
+  token is always valid for at least 26-28 days.
+   */
+  startTokenRefreshInterval() {
+    const twelveHours = 12 * 60 * 60 * 1e3
+    // Add some randomness to the interval so we
+    // can't correlate when a user is active/inactive quite as easily
+    const randomness = Math.random() * twelveHours
+    setInterval(() => {
+      void this.getTokenMaybeRefreshAndStoreState()
+    }, twelveHours + randomness)
```

Comment:
this isn't a very big deal, but it might be better to have a random delay after each refresh rather than calculating a random interval and maintaining it for the lifetime of the extension regularly

Reply:
yes, definitely